renku/models/cwl/command_line_tool.py

```diff
@@ -258,54 +258,76 @@ def watch(self, client, no_output=False, outputs=None):
         if repo:
             # List of all output paths.
             paths = []
-            # Keep track of unmodified output files.
-            unmodified = set()
-            # Possible output paths.
-            candidates = set(repo.untracked_files)
-            candidates |= {
-                item.a_path
-                for item in repo.index.diff(None) if not item.deleted_file
-            }
-
-            from renku.cli._graph import _safe_path
-            candidates = {path for path in candidates if _safe_path(path)}
 
             inputs = {input.id: input for input in self.inputs}
             outputs = list(tool.outputs)
 
-            for output, input, path in self.guess_outputs(candidates):
-                outputs.append(output)
-                paths.append(path)
+            if self.explicit_outputs:
```

Comment:
True! I am not sure why I went with this approach. It seems trivial to support both. Should I do that?

Reply:
I think it already works that way - maybe it has to do with the changes from @ableuler?

Reply:
It doesn't; I've just tried an example and it ignores generated outputs when `--output` is used.
Change from @ableuler has nothing to do with this. It was like this from the beginning.
I'll try a quick fix and if it work I'll commit the changes. Otherwise, we can decide what to do.

Reply:
Actually, it does it only for stdout and stderr outputs and not the rest. This is a bug that I fixed in the new commit.
Now, it's possible to have explicit outputs along with normal ones.